app.js

```diff
@@ -28,22 +28,17 @@ app.set('etag', false);
 app.set('trust proxy', true);
 app.set('query parser', querystring.parse);
 
-app.get(['/readiness_check', '/liveness_check', '/_ah/health'], function(
-  req,
-  res
-) {
+app.get(['/readiness_check', '/liveness_check', '/_ah/health'], (req, res) => {
   res.sendStatus(statusCodes.OK);
 });
 
-app.get('/r', (req, res) => {
-  return errorTracker(req, res, req.query);
-});
-app.post('/r', json, (req, res) => {
+app.get('/r', errorTracker);
+app.post('/r', json, async (req, res) => {
   // Allow non-credentialed posts from anywhere.
   // Not strictly necessary, but it avoids an error being reported by the
   // browser.
   res.set('Access-Control-Allow-Origin', '*');
-  return errorTracker(req, res, req.body);
+  return await errorTracker(req, res);
```

Comment:
Nit: Just `return await foo` is 100% equivalent to `return foo`.
```suggestion
  return errorTracker(req, res);
```

Reply:
Oh really, interesting. I was under the impression that would make the return type `Promise<Promise<T>>`. Done.

Reply:
No, it's impossible to have a promise of a promise, due to [promise adoption](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/resolve).